modules/integration/tests-common/admin-clients/src/main/java/org/wso2/mb/integration/common/clients/AndesClient.java

```diff
@@ -1,589 +1,317 @@
+     */
+    public long getSentMessageCount() {
+        long allSentMessageCount = 0L;
+        for (AndesJMSPublisher publisher : publishers) {
+            allSentMessageCount = allSentMessageCount + publisher.getSentMessageCount();
+        }
+        return allSentMessageCount;
     }
 
-    public boolean checkIfMessagesAreInOrder() {
-        AndesClientOutputParser andesClientOutputParser = new AndesClientOutputParser(filePathToWriteReceivedMessages);
-        return andesClientOutputParser.checkIfMessagesAreInOrder();
+    /**
+     * Gets the average transactions per seconds for publisher(s). Suppressing "UnusedDeclaration"
+     * as the client acts as an service.
+     *
+     * @return the average transactions per seconds.
+     */
+    @SuppressWarnings("UnusedDeclaration")
+    public double getPublisherTPS() {
+        double tps = 0L;
+        for (AndesJMSPublisher publisher : publishers) {
+            tps = tps + publisher.getPublisherTPS();
+        }
+        return tps / publishers.size();
     }
 
     /**
-     * This method return whether received messages are transacted
+     * Gets the duplicated messages received
      *
-     * @param operationOccurredIndex Index of the operated message most of the time last message
-     * @return
+     * @return A map of message identifiers and message content.
+     * @throws IOException
      */
-    public boolean transactedOperation(long operationOccurredIndex) {
-        AndesClientOutputParser andesClientOutputParser = new AndesClientOutputParser(filePathToWriteReceivedMessages);
-        return andesClientOutputParser.transactedOperations(operationOccurredIndex);
+    public Map<Long, Integer> checkIfMessagesAreDuplicated()
+            throws IOException {
+        if (0 < consumers.size()) {
+            AndesClientUtils.flushPrintWriters();
+            AndesClientOutputParser andesClientOutputParser =
+                    new AndesClientOutputParser(consumers.get(0).getConfig()
+                                                        .getFilePathToWriteReceivedMessages());
+            return andesClientOutputParser.getDuplicatedMessages();
+        } else {
+            return null;
+        }
     }
 
     /**
-     * This method returns number of duplicate received messages
+     * Checks whether the received messages are in order.
      *
-     * @return duplicate message count
+     * @return true if messages are in order, false otherwise.
+     * @throws IOException
      */
-    public int getTotalNumberOfDuplicates() {
-        AndesClientOutputParser andesClientOutputParser = new AndesClientOutputParser(filePathToWriteReceivedMessages);
-        return andesClientOutputParser.numberDuplicatedMessages();
+    public boolean checkIfMessagesAreInOrder()
+            throws IOException {
+        if (0 < consumers.size()) {
+            AndesClientOutputParser andesClientOutputParser =
+                    new AndesClientOutputParser(consumers.get(0).getConfig()
```

Comment:
The purpose was to check if there are any consumers in the "consumers" list. It would only be checking the first consumer's messages are in order. Would it be good if I check as this ? "if (!consumers.isEmpty())"